sktime/forecasting/hf_patchtst_forecaster.py

```diff
@@ -0,0 +1,507 @@
+
+    class Dataset:
+        """Dummy class if torch is unavailable."""
+
+        pass
+
+
+if _check_soft_dependencies("transformers", severity="none"):
+    from transformers import (
+        PatchTSTConfig,
+        PatchTSTForPrediction,
+        PatchTSTModel,
+        Trainer,
+        TrainingArguments,
+    )
+
+
+class HFPatchTSTForecaster(_BaseGlobalForecaster):
+    """Interface for the PatchTST forecaster.
+
+    Paper: https://arxiv.org/abs/2211.14730
+
+    This model has 3 available modes:
+        1) Loading an untrained PatchTST model and pretrain on some dataset.
+        This can be done by passing in a PatchTST config dictionary or passing
+        in available parameters in this estimator, or by loading the
+        HFPatchTSTForecaster without any passed arguments
+        2) Load a pre-trained model for fine-tuning. The parameter `model_path`
+        can be load to load a pre-trained model for fine-tuning. Using a
+        pretrained model will override any other model config parameters, and
+        you can use the `fit` function to then fine-tune your pretrained model.
+        See the `model_path` docstring for more details.
+        3) Load a pre-trained model for zero-shot forecasting. The parameter
+        `model_path` can be load to load a pre-trained model for fine-tuning.
+        Using a pretrained model will override any other model config
+        parameters, and you can use the `predict` function to do zero-shot
+        forecasting. See the `model_path` docstring for more details.
+
+    Parameters
+    ----------
+    model_path : str, optional
+        Path to the Huggingface model to use for global forecasting. If
+        model_path is passed, the remaining model config parameters will be
+        ignored except for specific training or dataset parameters.
+        This has 3 options:
+            - model id to an online pretrained PatchTST Model hosted on HuggingFace
+            - A path to a *directory* containing a configuration file saved
+            using the [`~PretrainedConfig.save_pretrained`] method,
+            or the [`~PreTrainedModel.save_pretrained`] method
+            - A path or url to a saved configuration JSON *file*, e.g.,
+                `./my_model_directory/configuration.json`.
+            forecast_columns,
+    mode : str, optional, possible values = ["untrained", "finetune", "zeroshot"]
+           default = "untrained"
+        String to set the mode of the model. If set to 'untrained', it will
+        re-initialize an untrained model with the specified config or
+        estimator aruguments. If set to "finetune" will use the `model_path`
+        argument and the passed in `y` in fit to fine-tune the model. If
+        set to "zeroshot", it will load the model in zero-shot forecasting
+        model with the argument `model_path` and ignore any passed `y`.
+        Note that both "finetune" and "zeroshot" mode requires a mandatory
+        passed in `model_path`.
```

Comment:
from what i understand:
- when `mode="untrained"` model is intiailized with random weights and trained using `y`
- when `mode="finetune"`, model is intiailized with pre-trained weights and trained using `y`
- when `mode="zero-shot"`, model is intiailized with pre-trained weights and is not trained

these training options are slightly different from the previous pre-trained models we have. I like these options but I think we would want to keep the fit strategies coherent with different pre-trained models. Do you have any comments about that?

can you please make the description more clear, I had to see the code to understand this, also make it more readable by putting this in bullets.

Reply:
I will make it more clear, thanks for giving your feedback and having a look :)

Reply:
> these training options are slightly different from the previous pre-trained models we have. I like these options but I think we would want to keep the fit strategies coherent with different pre-trained models. Do you have any comments about that?

This is something that I've noticed as well, and brought up on a thread on discord. I think there is no time series foundational model that currently supports all three 'modes'. So I've opted to create a 'mode' parameter to allow this. 

Would it be possible to refactor the 'mode' naming convention to align more with the 'fit' strategies with the other pre-trained models but keep the same functionalty? What do you think?

Reply:
we need a generic naming convention and a comparable functionality between different pre-trained models. I'll create an issue to track this and hopefully we can come to a solution before this PR gets merged so we don't end up with deprecation cycles for patchtst later.

Reply:
From collecting points in https://github.com/sktime/sktime/issues/7426, I think for `PatchTST`, this param should be called `fit_strategy` and have values between: `full`, `minimal` and `zero-shot`